velox/docs/develop/aggregate-functions.rst

```diff
@@ -109,25 +109,323 @@ accordingly.  Data is stored in the following order:
 .. image:: images/aggregation-layout.png
   :width: 600
 
-Aggregate class
----------------
-
-To add an aggregate function,
+To add an aggregate function, there are two options: implementing it as a
+simple function or as a vector function. The simple-function interface allows
+the author to write methods that process input data one-row-at-a-time and not
+handle input vector encodings themselves. However, the simple-function
+interface currently has certain limitations, such as not allowing for advanced
+performance optimization on constant inputs. Aggregation functions that
+require such functionalities can be implemented through the vector-function
+interface. With the vector-function interface, the author writes methods that
+process one-vector-at-a-time and handles input vector encodings by themselves.
 
 * Prepare:
     * Figure out what are the input, intermediate and final types.
     * Figure out what are partial and final calculations.
     * Design the accumulator. Make sure the same accumulator can accept both raw
       inputs and intermediate results.
-    * Create a new class that extends velox::exec::Aggregate base class
+    * If implementing a simple function, create a class for the function according
+      to instructions below; If implementing a vector function,
+      create a new class that extends velox::exec::Aggregate base class
       (see velox/exec/Aggregate.h) and implement virtual methods.
 * Register the new function using exec::registerAggregateFunction(...).
 * Add tests.
 * Write documentation.
 
-Accumulator size
+Simple Functions
```

Comment:
I changed it to "Simple Function Interface". I think "for Aggregates" is not needed because this entire documentation is already about aggregation functions.